edx_adapt/api/resources/course_resources.py

```diff
@@ -12,78 +13,56 @@
 course_parser.add_argument('course_id', type=str, required=True, location='json', help="Please supply a course ID")
 
 
-class Courses(Resource):
-    def __init__(self, **kwargs):
-        self.repo = kwargs['data']
-        """@type repo: DataInterface"""
-
-    def get(self):
-        courses = []
+class BaseResources(BaseResource):
```

Comment:
One cannot be constructed as generic for all resources modules, but you're that classes names look confusing will rename one of them.